File: db/role.php
<?php

require_once("entity.php");

/**
 * Class for accessing, manipulating, creating and deleting records in role.
*/

class Role extends Entity
{

  public function __construct($select = array())
  {
    $this->table = "role";
    $this->order = "rank";
    $this->domain = "role";
    $this->field['role_id']['type'] = 'SERIAL';
    $this->field['role_id']['not_null'] = true;
    $this->field['tag']['type'] = 'VARCHAR';
    $this->field['tag']['length'] = 32;
    $this->field['tag']['not_null'] = true;
    $this->field['rank']['type'] = 'INTEGER';
    $this->field['role_id']['primary_key'] = true;
    parent::__construct($select);
  }

}

?>

File: pentabarf/lib/momomoto/tables/role.rb
module Momomoto
  class Role < Base
    def initialize
      super
      @domain = 'roles'
      @fields = {
        :role_id => Datatype::Integer.new( {:not_null=>true, :default=>true, :primary_key=>true, :serial=>true} ),
        :tag => Datatype::Varchar.new( {:not_null=>true, :length=>32} ),
        :rank => Datatype::Integer.new( {} )
      }
    end
  end
end
